src/wp-includes/class-wp-modules.php

```diff
@@ -0,0 +1,299 @@
+<?php
+/**
+ * Modules API: WP_Modules class.
+ *
+ * Native support for ES Modules and Import Maps.
+ *
+ * @package WordPress
+ * @subpackage Modules
+ */
+
+/**
+ * Core class used to register modules.
+ */
+class WP_Modules {
```

Comment:
I know it's explained a little bit in the PR description, but it would be good to have a discussion about `WP_Dependencies` and `_WP_Dependency` and how these all fit together.

Reply:
Let's discuss that in the [ticket](https://core.trac.wordpress.org/ticket/56313).